ruma-client-api/src/r0/media/get_content.rs

```diff
@@ -59,8 +59,12 @@ impl<'a> Request<'a> {
     }
 
     /// Creates a new `Request` with the given url.
-    pub fn from_url(url: &'a MxcUri) -> Self {
-        Self { media_id: url.media_id(), server_name: url.server_name(), allow_remote: true }
+    pub fn from_url(url: &'a MxcUri) -> Result<Self, Error> {
+        if let (Some(media_id), Some(server_name)) = (url.media_id(), url.server_name()) {
```

Comment:
I think a method on `MxcUri` would make this nicer:

```suggestion
        if let Some((server_name, media_id)) = url.parts() {
```